modules/javafx.controls/src/test/java/test/javafx/scene/control/TableCellTest.java

```diff
@@ -402,6 +415,137 @@ public void testCellInUneditableColumnIsNotEditable() {
         assertFalse(cell.isEditing());
     }
 
+//------------ testing editCancel event: location on event - JDK-8187229
+
+    /**
+     * Basic config of table-/cell to allow testing of editEvents:
+     * table has editingColumn and cell is configured with table and column.
+     */
+    private void setupForEditing() {
+        table.setEditable(true);
+        table.getColumns().add(editingColumn);
+        // FIXME: default cell (of tableColumn) needs not-null value for firing cancel
+        editingColumn.setCellValueFactory(cc -> new SimpleObjectProperty<>(""));
+
+        cell.updateTableView(table);
+        cell.updateTableColumn(editingColumn);
+    }
+
+
+    /**
+     * Passes before fix of JDK-8187229, must pass after.
+     */
+    @Test
+    public void testEditCancelEventAfterCancelOnCell() {
+        setupForEditing();
+        int editingIndex = 1;
+        cell.updateIndex(editingIndex);
+        table.edit(editingIndex, editingColumn);
+        TablePosition<?, ?> editingPosition = table.getEditingCell();
+        List<CellEditEvent<?, ?>> events = new ArrayList<>();
+        editingColumn.setOnEditCancel(events::add);
+        cell.cancelEdit();
+        assertEquals("column must have received editCancel", 1, events.size());
+        assertEquals("editing location of cancel event", editingPosition, events.get(0).getTablePosition());
+    }
+
+    /**
+     * Fails before fix of JDK-8187229.
+     */
+    @Test
+    public void testEditCancelEventAfterCancelOnTable() {
```

Comment:
Minor: I think one more sentence about what to expect here would be good (so what is tested against here). Can be copied to all similar tests as well.